src/Symfony/Component/Webhook/Client/RequestParser.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Webhook\Client;
+
+use Symfony\Component\HttpFoundation\ChainRequestMatcher;
+use Symfony\Component\HttpFoundation\HeaderBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestMatcher\IsJsonRequestMatcher;
+use Symfony\Component\HttpFoundation\RequestMatcher\MethodRequestMatcher;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+use Symfony\Component\RemoteEvent\RemoteEvent;
+use Symfony\Component\Webhook\Exception\RejectWebhookException;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 6.3
+ */
+class RequestParser extends AbstractRequestParser
+{
+    protected function getRequestMatcher(): RequestMatcherInterface
+    {
+        return new ChainRequestMatcher([
+            new MethodRequestMatcher('POST'),
+            new IsJsonRequestMatcher(),
+        ]);
+    }
+
+    protected function doParse(Request $request, string $secret): RemoteEvent
+    {
+        // $body = $this->bodyParser->parse($request, $secret);
+        $body = $request->toArray();
+
+        // $this->headerParser->validate($request, $secret);
+        foreach (['x-id', 'x-event', 'x-signature'] as $header) {
+            if (!$request->headers->has($header)) {
+                throw new RejectWebhookException(406, sprintf('Missing "%s" HTTP request signature header.', $header));
+            }
+        }
+
+        // $this->signatureParser->validate($request, $secret);
+        $this->validateSignature($request->headers, $request->getContent(), $secret);
+
+        return new RemoteEvent($request->headers->get('x-event'), $request->headers->get('x-id'), $body);
+    }
+
+    private function validateSignature(HeaderBag $headers, string $body, $secret): void
+    {
+        if (!hash_equals($headers->get('x-signature'), 'sha256='.hash_hmac('sha256', $headers->get('x-event').$headers->get('x-id').$body, $secret))) {
```

Comment:
if this class is meant to be used as the counter part of the classes from the `Server` namespace, I think it would be a good idea to make the header name and the signature algorithm configurable in the same way